packages/haunted-hooks-testing-library/test/main.ts

```diff
@@ -8,44 +8,374 @@ import {
+      expect(result.current?.counter).toBe(0);
+    });
+
+    // use useRef to test this one, we can't trigger updates
+    test("resolve during interval with no updates", async () => {
+      function hook() {
+        const counter = useRef(0);
+
+        function increment() {
+          setTimeout(() => {
+            counter.current++;
+            increment(); // this leaks memory because no TCO in Node
+          }, 10);
+        }
+
+        increment();
+
+        return {
+          counter,
+        };
+      }
+
+      const { result, waitFor } = renderHook(hook);
+
+      expect(result.current?.counter.current).toBe(0);
+
+      await waitFor(() => result.current?.counter.current === 2, {
+        interval: 2,
+      });
+      expect(result.current?.counter.current).toBe(2);
+
+      await waitFor(() => expect(result.current?.counter.current).toBe(4), {
+        interval: 2,
+      });
+      expect(result.current?.counter.current).toBe(4);
+    });
+  });
+
+  describe("waitForValueToChange", () => {
+    test("resolve when value changes", async () => {
+      function hook() {
+        const [counter, setCounter] = useState(0);
+
+        setTimeout(() => setCounter((c = 0) => c + 1), 10);
+
+        return {
+          counter,
+          setCounter,
+        };
+      }
+
+      const { result, waitForValueToChange } = renderHook(hook);
+
+      expect(result.current?.counter).toBe(0);
+
+      await waitForValueToChange(() => result.current?.counter === 2);
+      expect(result.current?.counter).toBe(2);
+
+      await waitForValueToChange(() => result.current?.counter);
+      expect(result.current?.counter).toBe(3);
+    });
+
+    //*
+    test("timeout expiry", async () => {
+      let error: Error | undefined = undefined;
+
+      function hook() {
+        const [counter, setCounter] = useState(0);
+
+        setTimeout(() => setCounter((c = 0) => c + 1), 10);
+
+        return {
+          counter,
+          setCounter,
+        };
+      }
+
+      const { result, waitForValueToChange } = renderHook(hook);
+
+      expect(result.current?.counter).toBe(0);
+
+      await waitForValueToChange(() => result.current?.counter === 1);
+      expect(result.current?.counter).toBe(1);
+
+      try {
+        await waitForValueToChange(() => result.current?.counter, {
+          timeout: 1,
+        });
+      } catch (e) {
+        error = e;
+      }
+      expect(error?.name).toBe("TimeoutError");
+      expect(result.current?.counter).toBe(1);
+
+      await waitForValueToChange(() => result.current?.counter);
+      expect(result.current?.counter).toBe(2);
+    });
+
+    test("default timeout", async () => {
+      let error: Error | undefined = undefined;
+
+      function hook() {
+        const [counter, setCounter] = useState(0);
+
+        setTimeout(() => setCounter((c = 0) => c + 1), 2000);
+
+        return {
+          counter,
+          setCounter,
+        };
+      }
```

Comment:
This code could be way more DRY, just add a `timeout` prop to `hook`, and use it in all the tests

Reply:
Done. Did the same thing with the useRef hook.